tests/generic_config_updater/test_eth_interface.py

```diff
@@ -237,10 +233,43 @@ def test_update_valid_invalid_index(duthost, ensure_dut_readiness, index, is_val
 
     try:
         output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
-        if is_valid:
-            expect_op_success(duthost, output)
-        else:
-            expect_op_failure(output)
+        expect_op_failure(output)
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+def test_update_valid_index(duthost, ensure_dut_readiness):
+    output = duthost.shell('sonic-db-cli CONFIG_DB keys "PORT|"\\*')["stdout"]
+    interfaces = {}  # to be filled with two interfaces mapped to their indeces
+
+    for line in output.split('\n'):
+        if line.startswith('PORT|Ethernet'):
+            interface = line[line.index('Ethernet'):].strip()
+            index = duthost.shell('sonic-db-cli CONFIG_DB hget "PORT|{}" index'.format(interface))["stdout"]
+            interfaces[interface] = index
```

Comment:
This part can be done by running `sonic-cfggen -d --var-json "PORT"`. The output is a json, including port names and index.